src/components/button/button.component.js

```diff
@@ -29,6 +29,7 @@ function renderChildren({
   };
 
   const iconProps = {
+    "aria-hidden": !!children,
```

Comment:
I think that it could be always hidden, as the icon in button should always be ignored by readers

Reply:
On further discussion, `Icon` should always be hidden and `aria-label` should be set to `Button` instead. This is to avoid being able to focus on the inner icon.